### PR TITLE
 Add criupath testcase and refactor code

### PR DESCRIPTION
**Pull request by chchliang**
Signed-off-by: chchliang <chen.chuanliang@zte.com.cn>

### CODE REVIEW

File: libcontainer/factory_linux_test.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ func TestFactoryNew(t *testing.T) {
 	if factory.Type() != "libcontainer" {
 		t.Fatalf("unexpected factory type: %q, expected %q", factory.Type(), "libcontainer")
 	}
+
+	if lfactory.CriuPath != "criu" {
+		t.Fatalf("unexpected factory CriuPath: %q, expected %q", lfactory.CriuPath, "criu")
```
Comment (cyphar) [inline libcontainer/factory_linux_test.go]: What is the reason for this change?

Comment (chchliang) [inline libcontainer/factory_linux_test.go]: I think `defer syscall.Unmount(root, syscall.MNT_DETACH)` at last line of this function is wrong,by the way add `lfactory.CriuPath` check, like `factory.Type() != "libcontainer"`

**Comment by chchliang** [comment]
@cyphar 
 I have refactored this code, `func checkFactory(factory Factory, root string) (*LinuxFactory, error) `
